src/lib/optimizer/chunk_statistics/range_filter.hpp

```diff
@@ -26,16 +26,37 @@ class RangeFilter : public AbstractFilter {
   explicit RangeFilter(std::vector<std::pair<T, T>> ranges) : _ranges(std::move(ranges)) {}
   ~RangeFilter() override = default;
 
-  static std::unique_ptr<RangeFilter<T>> build_filter(const pmr_vector<T>& dictionary);
+  static std::unique_ptr<RangeFilter<T>> build_filter(const pmr_vector<T>& dictionary,
+                                                      uint32_t max_ranges_count = MAX_RANGES_COUNT);
 
   bool can_prune(const AllTypeVariant& value, const PredicateCondition predicate_type) const override {
     const auto t_value = boost::get<T>(value);
+    // Operators work as follows: value_from_table <operator> t_value
+    // e.g. OpGreaterThan: value_from_table > t_value
+    // thus we can exclude chunk if t_value >= _max since then no value from the table can be greater than t_value
     switch (predicate_type) {
+      case PredicateCondition::GreaterThan: {
+        auto& max = _ranges.back().second;
+        return t_value >= max;
+      }
+      case PredicateCondition::GreaterThanEquals: {
+        auto& max = _ranges.back().second;
+        return t_value > max;
+      }
+      case PredicateCondition::LessThan: {
+        auto& min = _ranges.front().first;
+        return t_value <= min;
+      }
+      case PredicateCondition::LessThanEquals: {
+        auto& min = _ranges.front().first;
+        return t_value < min;
+      }
       case PredicateCondition::Equals: {
-        bool prunable = false;
-        for (const auto& min_max_pair : _ranges) {
-          const auto & [ min, max ] = min_max_pair;
-          prunable |= min < t_value && t_value < max;
+        bool prunable = true;
+        for (const auto& bounds : _ranges) {
+          const auto & [ min, max ] = bounds;
+          // prunable becomes false if t_value is within any of the bounds
```

Comment:
I guess this comment is outdated, right?

Reply:
No, i think its up to date. We are storing ranges where values exist, so when a value falls into a range there is the possibilty that the value is actually contained in the column and `Equals` will produce results.

Reply:
Yes, I updated the comment.